cluster-provision/k8s/1.19/bind_device_to_vfio.sh

```diff
@@ -50,6 +50,10 @@ fi
 # The device should now be using the vfio-pci driver
 new_driver=$(get_device_driver)
 if [[ $new_driver != "vfio-pci" ]]; then
-    echo "Error: Failed to bind to vfio-pci driver" 1>&2
-    exit 1
+    sleep 10
+    new_driver=$(get_device_driver)
+    if [[ $new_driver != "vfio-pci" ]]; then
+        echo "Error: Failed to bind to vfio-pci driver" 1>&2
+        exit 1
+    fi
 fi
```

Comment:
it might work, (similar solutions were done in the past)
maybe worth to use timeout until ... sleep 1 instead ? 
see cluster-up/cluster/kind/check-cluster-up.sh for an example please

Reply:
I think it's going to look worse because instead of blocking until it's true, we fail if it isn't.
so it will look like
timeout bash -c 'until [[ long-conditions ]]; do sleep 1; done || echo "failed" && exit
or something like that. it looks like a really long complicated line :)

Reply:
you dont need the exit,
the timeout will give an error, so scripts with -e will fail directly (in case script here is with -e)
`timeout -s10 bash -c 'until [[ long-conditions ]]; do sleep 1; done`
u need just this (added the -s10 or whatever the syntax is)

Reply:
so I tried this:
`timeout -v 10s bash -c "until [ x$(get_device_driver) = "xvfio-pci" ]; do sleep 1; done"`
and I'm not super sure about order of evaluating variables in bash because the output looks like this:
```
+ timeout -v 10s bash -c 'until [ x = xvfio-pci ]; do sleep 1; done'
timeout: sending signal TERM to command ‘bash’
```
It might only run this function once, and not have the desired effect of re-testing.

(I am not sure it's worth pursuing it further)

Reply:
$ timeout 5s bash -c "until false; do echo bla; sleep 1; done"
bla
bla
bla
bla
bla


it does repeat this way